scipy/diff/_derivative_numdiff.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import division
+import numpy as np
+from _step_generators import _generate_step
+from scipy import misc
+from scipy.ndimage.filters import convolve1d
+from scipy import linalg
+import warnings
+
+EPS = np.finfo(float).eps
+TINY = np.finfo(float).tiny
+
+
+def choose_derivative_with_least_error(der, errors):
+    try:
+        median = np.nanmedian(errors, axis=0)
+        p75 = np.nanpercentile(errors, 75, axis=0)
+        p25 = np.nanpercentile(errors, 25, axis=0)
+        iqr = np.abs(p75-p25)
+        a_median = np.abs(median)
+        outliers = (((abs(errors) < (a_median / 10)) +
+                    (abs(errors) > (a_median * 10))) * (a_median > 1e-8) +
+                    ((errors < p25-1.5*iqr) + (p75+1.5*iqr < der)))
+        errors = errors + outliers * np.abs(errors - median)
+    except ValueError as msg:
```

Comment:
I am sorry for msg variable, I used msg while debugging and forgot to remove it.